tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/Session.java

```diff
@@ -141,6 +171,41 @@ public void close() {
     }
   }
 
+  /**
+   * Execute any un-ran initializers. Will be done automatically unless disabled at session
+   * creation.
+   *
+   * <p>This runs any ops that have been created with an init scope that have not already been ran.
+   */
+  public synchronized void initialize() {
+    Runner runner = runner();
+    synchronized (graph) {
+      graph.initializers().stream().filter((x) -> !ranInits.contains(x)).forEach(runner::addTarget);
+      ranInits = graph.initializers();
+    }
+    if (!runner.isEmpty()) {
+      runner.runNoInit();
+    }
+  }
+
+  /**
+   * Execute the graph's initializers, regardless of whether the session has been initialized.
+   *
+   * <p>This runs any ops that have been created with an init scope.
+   *
+   * @return this
+   */
+  public synchronized Session forceInitialize() {
```

Comment:
It's if you need to reset the session state.  Not something you should be using probably, but there just in case, since there's no way to do it otherwise.